src/scores/categorical/contingency_impl.py

```diff
@@ -234,6 +279,142 @@ def specificity(self):
+        Equitability revisited: Why the “equitable threat score” is not equitable.
+        Weather and Forecasting, 25(2), pp.710-726.
+        """
+        cd = self.counts
+        hits_random = (cd["tp_count"] + cd["fn_count"]) * (cd["tp_count"] + cd["fp_count"]) / cd["total_count"]
+        ets = (cd["tp_count"] - hits_random) / (cd["tp_count"] + cd["fn_count"] + cd["fp_count"] - hits_random)
+
+        return ets
+
+    def gilberts_skill_score(self):
+        """
+        Calculates the Gilbert skill score (also known as the Equitable threat score).
+
+        How well did the forecast "yes" events correspond to the observed "yes"
+        events (accounting for hits due to chance)?
+
+        Range: -1/3 to 1, 0 indicates no skill. Perfect score: 1.
+
+        Hogan, R.J., Ferro, C.A., Jolliffe, I.T. and Stephenson, D.B., 2010.
+        Equitability revisited: Why the “equitable threat score” is not equitable.
+        Weather and Forecasting, 25(2), pp.710-726.
+        """
+        return self.equitable_threat_score()
+
+    def heidke_skill_score(self):
+        """
+        Calculates the Heidke skill score (also known as Cohen's kappa).
+
+        What was the accuracy of the forecast relative to that of random chance?
+
+        Range: -1 to 1, 0 indicates no skill. Perfect score: 1.
+
+        https://en.wikipedia.org/wiki/Cohen%27s_kappa
+        """
+        cd = self.counts
+        exp_correct = (1 / cd["total_count"]) * (
+            (cd["tp_count"] + cd["fn_count"]) * (cd["tp_count"] + cd["fp_count"])
+            + ((cd["tn_count"] + cd["fn_count"]) * (cd["tn_count"] + cd["fp_count"]))
+        )
+        hss = ((cd["tp_count"] + cd["tn_count"]) - exp_correct) / (cd["total_count"] - exp_correct)
+        return hss
+
+    def cohens_kappa(self):
+        """
+        Calculates the Cohen's kappa (also known as the Heidke skill score).
+
+        What was the accuracy of the forecast relative to that of random chance?
+
+        Range: -1 to 1, 0 indicates no skill. Perfect score: 1.
+
+        https://en.wikipedia.org/wiki/Cohen%27s_kappa
+        """
+        return self.heidke_skill_score()
+
+    def odds_ratio(self):
+        """
+        Calculates the odds ratio
+
+        What is the ratio of the odds of a "yes" forecast being correct, to the odds of
+        a "yes" forecast being wrong?
+
+        Odds ratio - Range: 0 to ∞, 1 indicates no skill. Perfect score: ∞.
+
+        Stephenson, D.B., 2000. Use of the “odds ratio” for diagnosing forecast skill.
+        Weather and Forecasting, 15(2), pp.221-232.
+        """
+        odds_r = (self.probability_of_detection() / (1 - self.probability_of_detection())) / (
+            self.probability_of_false_detection() / (1 - self.probability_of_false_detection())
+        )
+        return odds_r
+
+    def odds_ratio_skill_score(self):
+        """
+        Calculates the odds ratio skill score (also known as Yule's Q).
```

Comment:
I think we should eventually have a whole separate API for skill scores which compare between alternative predictions, where any base metric can be used to underpin a skill score calculation. If there are skill scores that aren't simple ratios between scores, we can also add them there. This metric doesn't seem to be a skill score in the sense of comparing the skill between alternative predictions. This metric does look like a score or metric. Given it is called what it is, for better or worse, it makes sense to have here. I will add something to the docstring to briefly explain this may not match common use of the term 'skill score'. I will raise an issue matching this comment as a reminder to consider this issue post-merge as well as review the change I make to the docstring.